internal/feedback/feedback.go

```diff
@@ -82,6 +82,55 @@ func (cf ConstraintFeedback) LogFeedback(logger *log.Logger) {
 	}
 }
 
+type brokenImport struct {
+	projectPath string
+	version     *gps.StringDiff
+	revision    *gps.StringDiff
+}
+
+func (bi brokenImport) String() string {
+	var pr string
+	var cr string
+	if bi.revision != nil {
+		pr = fmt.Sprintf("(%s)", trimSHA(bi.revision.Previous))
+		cr = fmt.Sprintf("(%s)", trimSHA(bi.revision.Current))
+	}
+
+	return fmt.Sprintf("%v %s for %s. Locking in %v %s",
+		bi.version.Previous,
+		pr,
+		bi.projectPath,
+		bi.version.Current,
+		cr,
+	)
+}
+
+// BrokenImportFeedback holds problematic lock feedback data
+type BrokenImportFeedback struct {
+	brokenImports []brokenImport
+}
+
+// NewBrokenImportFeedback builds a feedback entry for problems with imports from a diff of the pre- and post- solved locks
+func NewBrokenImportFeedback(ld *gps.LockDiff) *BrokenImportFeedback {
+	bi := &BrokenImportFeedback{}
+	for _, lpd := range ld.Modify {
```

Comment:
I wasn't sure if we also needed to cover `Add` and `Remove` here.  As of now, this is only taking note of `Modify` from the `LockDiff`.

Reply:
We also want to warn when the final lock has _removed_ a project. Maybe something like:

```
Warning: Unable to preserve imported lock v1.1.4 (bc29b4f) for github.com/foo/bar. The project was removed from the lock because it is not used.
```

For example, I had a project in my glide.lock that was imported, but was then removed by dep. It may help me realize that I need to manually add a `required` entry for that project to my manifest, to force it to be included.

We could look for adds, but that is beyond the scope of broken locks, and doesn't belong in this PR. That is what the existing functionality is doing in the else block of FinalizeManifestAndLock.